codalab/server/bundle_manager.py

```diff
@@ -570,6 +613,20 @@ def _filter_and_sort_workers(self, workers_list, bundle, bundle_resources):
                 for worker in workers_list
                 if not worker['tag_exclusive'] or not worker['tag']
             ]
+            if not workers_list:
+                staged_status = 'No workers available. Try connecting a worker to the CodaLab instance and using that to run this bundle.'
+                self._set_staged_status(bundle, staged_status)
+                return []
+
+        # Filter by the number of jobs allowed to run on this worker.
+        workers_list = [worker for worker in workers_list if worker['exit_after_num_runs'] > 0]
```

Comment:
Are there workers with exit_after_num_runs = 0?  This shouldn't be the case?

Reply:
Seems like a possible edge case: https://github.com/codalab/codalab-worksheets/blob/master/codalab/server/bundle_manager.py#L513